tests/ui/anon-params/anon-params-denied-2018.stderr

```diff
@@ -45,10 +45,14 @@ LL |     fn foo_with_qualified_path(<Bar as T>::Baz);
    |                                               ^ expected one of 8 possible tokens
    |
    = note: anonymous parameters are removed in the 2018 edition (see RFC 1685)
-help: explicitly ignore the parameter name
+help: if this is a `self` type, give it a parameter name
+   |
+LL |     fn foo_with_qualified_path(self: <Bar as T>::Baz);
+   |                                +++++
```

Comment:
Mmm... Arbitrary self types only support a handful of pointer types (`Box`, `Arc`, `Rc`). General arbitrary self types are unlikely to be supported any time soon, so adding this suggestion would be confusing to users. (I know, I know, but we have to try to minimize confusion to users, the only thing worse than an error without a suggestion is an error with a wrong suggestion.)

Reply:
As mentioned below (https://github.com/rust-lang/rust/pull/114300#discussion_r1282661709)